src/App.tsx

```diff
@@ -1,15 +1,49 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [dropDown, setDropDown] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 700), []);
```

Comment:
JFYI your variant is totally ok, but the following is OK too:

```suggestion
  const applyQuery = useCallback(debounce(setAppliedQuery, 700), [setAppliedQuery]);
```